tests/sqlAst/data/sql-ast-narrowing.php

```diff
@@ -7,6 +7,77 @@
 
 class Foo
 {
+    public function whereIsNotNull(PDO $pdo): void
```

Comment:
another thing just came to mind: I think we need another test which asserts the type of a column from a joined table narrowed via WHERE

Reply:
Good suggestion, I've rebased and added tests for this case. Thanks!